AdaptiveUiCodelab/app/src/main/java/com/example/reply/ui/ReplyHomeViewModel.kt

```diff
@@ -43,14 +43,27 @@ class ReplyHomeViewModel(private val emailsRepository: EmailsRepository = Emails
                     _uiState.value = ReplyHomeUIState(error = ex.message)
                 }
                 .collect { emails ->
-                    _uiState.value = ReplyHomeUIState(emails = emails)
+                    // If nothing is selected, initially select the first element.
+                    val currentSelected = _uiState.value.selectedEmail
+                    _uiState.value = ReplyHomeUIState(
+                        emails = emails,
+                        selectedEmail = currentSelected ?: emails.firstOrNull()
+                    )
                 }
         }
     }
+
+    fun setSelectedEmail(email: Email) {
+        val currentState = _uiState.value
+        _uiState.value = currentState.copy(
+            selectedEmail = email
+        )
```

Comment:
Nit: We could probably use [`MutableStateFlow.update`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-mutable-state-flow/#1996476265%2FFunctions%2F1975948010) here instead

Reply:
Nice one, done